proxy/coco/src/state.rs

```diff
@@ -443,6 +444,36 @@ impl State {
             })
             .collect()
     }
+
+    pub fn checkout<P>(
+        &self,
+        urn: &RadUrn,
+        peer_id: P,
+        destination: PathBuf,
+    ) -> Result<PathBuf, Error>
+    where
+        P: Into<Option<PeerId>>,
+    {
+        let proj = self.get_project(&urn, peer_id).map_err(Error::from)?;
+        let include = self.prepare_include(&proj).map_err(Error::from)?;
+        let checkout = project::Checkout::new(proj, destination, include);
+
+        checkout.run(self.peer_id()).map_err(Error::from)
+    }
+
+    pub fn prepare_include(
```

Comment:
```suggestion
    pub fn update_include(
```

We are going to call this for many project actions as per #930